sklearn/_build_utils/__init__.py

```diff
@@ -70,20 +71,37 @@ def cythonize_extensions(extension):
         os.environ.get("SKLEARN_ENABLE_DEBUG_CYTHON_DIRECTIVES", "0") != "0"
     )
 
+    compiler_directives = {
+        "language_level": 3,
+        "boundscheck": cython_enable_debug_directives,
+        "wraparound": False,
+        "initializedcheck": False,
+        "nonecheck": False,
+        "cdivision": True,
+    }
+
+    # TODO: once Cython 3 is released and we require Cython>=3 we should get
+    # rid of the `legacy_implicit_noexcept` directive.
+    # This should mostly consist in:
+    # 
```

Comment:
According to the [linting job]https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=50782&view=logs&jobId=32e2e1bb-a28f-5b18-6cfc-3f01273f5609), there is a black formatting issue here.